handlers/photos-tactile-svg/tactile_svg.py

```diff
@@ -0,0 +1,255 @@
+        request_schema = json.load(f)
+    with open("./schemas/handler-response.schema.json") as f:
+        response_schema = json.load(f)
+    with open("./schemas/renderers/tactilesvg.schema.json") as f:
+        renderer_schema = json.load(f)
+
+    store = {
+        definitions_schema["$id"]: definitions_schema,
+        request_schema["$id"]: request_schema,
+        response_schema["$id"]: response_schema,
+        renderer_schema["$id"]: renderer_schema,
+    }
+    resolver = jsonschema.RefResolver.from_schema(
+        request_schema, store=store
+    )
+    # Get and validate request contents
+    contents = request.get_json()
+    try:
+        validator = jsonschema.Draft7Validator(
+            request_schema, resolver=resolver
+        )
+        validator.validate(contents)
+    except ValidationError as e:
+        logging.error(e)
+        return jsonify("Invalid request received!"), 400
+
+    # Check preprocessor data
+    preprocessors = contents['preprocessors']
+
+    """
+    if "ca.mcgill.a11y.image.renderer.TactileSVG" not in contents["renderers"]:
+        logging.debug("TactileSVG Renderer not supported")
+        response = {
+            "request_uuid": contents["request_uuid"],
+            "timestamp": int(time.time()),
+            "renderings": []
+        }
+        try:
+            validator = jsonschema.Draft7Validator(
+                response_schema, resolver=resolver)
+            validator.validate(response)
+        except jsonschema.exceptions.ValidationError as error:
+            logging.error(error)
+            return jsonify("Invalid Preprocessor JSON format"), 500
+        logging.debug("Sending response")
+        return response
+    """
+
+    # No Object Detector AND semantic segmentation found
+    if not any(x in preprocessors for x in
+               ["ca.mcgill.a11y.image.preprocessor.semanticSegmentation",
+                "ca.mcgill.a11y.image.preprocessor.objectDetection"]):
+        logging.debug("No Object Detector and Semantic Segmentation found")
+        response = {
+            "request_uuid": contents["request_uuid"],
+            "timestamp": int(time.time()),
+            "renderings": []
+        }
+        try:
+            validator = jsonschema.Draft7Validator(
+                response_schema, resolver=resolver)
+            validator.validate(response)
+        except jsonschema.exceptions.ValidationError as error:
+            logging.error(error)
+            return jsonify("Invalid Preprocessor JSON format"), 500
+        logging.debug("Sending response")
+        return response
+
+    if "dimensions" in contents:
+        # If an existing graphic exists, often it is
+        # best to use that for convenience.
+        # see the following for SVG coordinate info:
+        # developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Positions
+        dimensions = contents["dimensions"]
+    else:
+        logging.debug("Dimensions are not defined")
```

Comment:
drawSvg library needs dimensions during the drawing initialization step. One alternative way to do this would be to obtain it from the graphic when not present but it's obviously a little more trouble